DQM/SiPixelPhase1Summary/src/SiPixelPhase1Summary.cc

```diff
@@ -92,21 +97,32 @@ void SiPixelPhase1Summary::dqmEndLuminosityBlock(DQMStore::IBooker & iBooker, DQ
 //------------------------------------------------------------------
 void SiPixelPhase1Summary::dqmEndJob(DQMStore::IBooker & iBooker, DQMStore::IGetter & iGetter)
 {
-  if (runOnEndJob_) fillSummaries(iBooker,iGetter);
+  if (firstLumi){ //Book the plots in the (maybe possible?) case that they aren't booked in the dqmEndLuminosityBlock method
+    bookSummaries(iBooker);
+    bookTrendPlots(iBooker);
+    firstLumi = false;
+  }
+  if (runOnEndJob_){
+    fillSummaries(iBooker,iGetter);
+    if (!runOnEndLumi_) fillTrendPlots(iBooker,iGetter); //If we're filling these plots at the end lumi step, it doesn't really make sense to also do them at the end job
+  }
 
 }
 
 //------------------------------------------------------------------
 // Used to book the summary plots
 //------------------------------------------------------------------
 void SiPixelPhase1Summary::bookSummaries(DQMStore::IBooker & iBooker){
-  iBooker.setCurrentFolder("PixelPhase1/Summary");
+
+  iBooker.cd();
 
   std::vector<std::string> xAxisLabels_ = {"BMO","BMI","BPO ","BPI","HCMO_1","HCMO_2","HCMI_1","HCMI_2","HCPO_1","HCPO_2","HCPI_1","HCPI_2"}; // why not having a global variable !?!?!?! 
-  std::vector<std::string> yAxisLabels_ = {"1","2","3","4"}; // why not having a global variable ?!?!?!!?
+  std::vector<std::string> yAxisLabels_ = {"1","2","3","4"}; // why not having a global variable ?!?!?!!? - I originally did, but was told not to by David Lange!
```

Comment:
first link on google might be informative - google can tell you more about why to avoid using globals in c++ (let alone a multiply threaded c++)
http://www.cplusplus.com/forum/general/76476/